sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -315,16 +325,18 @@ def __init__(self, key, secret_uri, label=None, **kwargs):
         self.last_modified = kwargs.get("last_modified", None)
         self.read_only = kwargs.get("read_only", None)
         self.tags = kwargs.get("tags", {})
-        self.value = {"secret_uri": self._secret_uri}
+        self.value = secret_id
+        if not isinstance(secret_id, dict):
+            self.value = {"secret_uri": secret_id}
 
     @property
-    def secret_uri(self):
+    def secret_id(self):
         # type: () -> str
         self._validate()
         return self.value['secret_uri']
 
-    @secret_uri.setter
-    def secret_uri(self, value):
+    @secret_id.setter
+    def secret_id(self, value):
         if self.value is None or isinstance(self.value, dict):
```

Comment:
```suggestion
        if self.value is None:
            self.value = {}
        if isinstance(self.value, dict):
            self.value["secret_uri"] = value
        else:
```

Reply:
Why do we want a nested dictionary here? The format in the portal is `{secret_uri: <secret_uri_value>}`.